include/net/net_ip.h

```diff
@@ -863,6 +863,22 @@ int net_addr_pton(sa_family_t family, const char *src, void *dst);
 char *net_addr_ntop(sa_family_t family, const void *src,
 		    char *dst, size_t size);
 
+/**
+ * @brief Compare TCP sequence numbers.
+ *
+ * @details This function compares TCP sequence numbers,
+ *          accounting for wraparound effects.
+ *
+ * @param seq1 First sequence number
+ * @param seq2 Seconds sequence number
+ *
+ * @return < 0 if seq1 < seq2, 0 if seq1 == seq2, > 0 if seq > seq2
+ */
+static inline s32_t net_tcp_seq_cmp(u32_t seq1, u32_t seq2)
```

Comment:
I do not think this will give correct answer in the wrapping case:

For example when using these values (seq2 is wrapped)
seq1 = UINT32_MAX - 10
seq2 = 10

would return <0 value even thou seq2 is "greater" than seq1.

Example program to test this:
#include <stdio.h>
#include <stdint.h>

signed int net_tcp_seq_cmp(unsigned int seq1, unsigned int seq2)
{
	return (signed int)(seq1 - seq2);
}

void main(void)
{
	int ret;
	unsigned int d1 = 10;
	unsigned int d2 = UINT32_MAX - 10;

	ret = net_tcp_seq_cmp(d1, d2);
	printf("1) d1 < d2 = %u < %u = %d (%d)\n", d1, d2, ret, ret > 0);

	ret = net_tcp_seq_cmp(d2, d1);
	printf("2) d1 < d2 = %u < %u = %d (%d)\n", d2, d1, ret, ret > 0);
}